pkg/aggregator/aggregator.go

```diff
@@ -384,6 +398,30 @@ func (agg *BufferedAggregator) handleSenderBucket(checkBucket senderHistogramBuc
 	}
 }
 
+func (agg *BufferedAggregator) handleEventPlatformEvent(event senderEventPlatformEvent) error {
+	agg.mu.Lock()
+	defer agg.mu.Unlock()
+	if agg.eventPlatformForwarder == nil {
+		return errors.New("event platform forwarder not initialized")
+	}
```

Comment:
We could do this before the lock to avoid locking when event platform is disabled).

Reply:
Good point, updated. 

Note that it will always be "enabled" by default the way this PR is currently written. As far as I can see it will only be "disabled" (i.e. `nil`) in tests.

Reply:
Good point, but if one day we change the behavior the lock logic will be ready.